server/stats.go

```diff
@@ -176,6 +257,38 @@ func triplesToStatsVar(
 	return &statsVar, nil
 }
 
+// getValue get the stat value from ObsTimeSeries.
+// When date is given, it get the value from the highest ranked source series
+// that has the date.
+// When date is not given, it get the latest value from the highest ranked
+// source series.
+func getValue(in *ObsTimeSeries, date string) (float64, error) {
```

Comment:
I'm trying to understand why this logic is separate and different from filterAndRank()?

Would it be better to have a common function for ranking series that takes optional mmethod, unit, period, date?

Reply:
Right now getValue does not care the mmethod etc since it is tailored for the simple api where user does not require knowledge on these. After we have the new apis ready, no api will take mmethod as argument, instead the /stat/all will return all series then client can pick from those.

After I can combine the logic together so the ranking score only appears in one place.

Reply:
SG